torch/nn/parallel/distributed.py

```diff
@@ -895,7 +895,7 @@ def forward(self, *inputs, **kwargs):
             # specified as part of hook, if hook was specified.
             buffer_hook_registered = hasattr(self, 'buffer_hook')
             if self.require_forward_param_sync and (not buffer_hook_registered or self.buffer_hook.buffer_comm_hook_location == _BufferCommHookLocation.PRE_FORWARD):
```

Comment:
nit: put this into a helper function, it could also include 'if self.will_sync_module_buffers()' check in this helper function

Reply:
e.g., will_sync_module_buffers() can include require_forward_param_sync check; should_sync_buffer_forward(), should_sync_buffer_backward()